Ladybird/Android/src/main/java/org/serenityos/ladybird/WebViewImplementation.kt

```diff
@@ -10,13 +10,18 @@ import android.content.Context
 import android.content.Intent
 import android.graphics.Bitmap
 import android.util.Log
+import android.view.View
+import java.net.URL
 
 /**
  * Wrapper around WebView::ViewImplementation for use by Kotlin
  */
 class WebViewImplementation(
     context: Context,
-    private var appContext: Context = context.applicationContext
+    view: View,
+    private var parentView: View = view,
+    private var appContext: Context = context.applicationContext,
+    private var resourceDir: String = (context as LadybirdActivity).resourceDir
```

Comment:
This is getting way out of control 😀
I don't think you need to pass anything but `view: WebView` here. Maybe `resourceDir` too, but the `LadybirdActivity.resourceDir` logic should live in the activity, the views shouldn't make too many assumptions about which activities they're a part of. In other words: there'd be a setter that the activity would call.

Reply:
It can't be in a setter, I need it in `nativeObjectInit`, which is called as part of the constructor of the object.

Reply:
well I guess I could move the call to nativeObjectInit to a `initialize()` function :thinking: